rectools/metrics/debias.py

```diff
@@ -0,0 +1,118 @@
+#  Copyright 2024 MTS (Mobile Telesystems)
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""Debias module."""
+
+import attr
+import pandas as pd
+
+from rectools import Columns
+
+
+@attr.s
+class DebiasConfig:
+    """
+    Config with debias method parameters.
+
+    Parameters
+    ----------
+    iqr_coef : float, default 1.5
+        Coefficient for defining as the maximum value inside the border.
+    random_state : float, default 32
+        Pseudorandom number generator state to control the down-sampling.
+    """
+
+    iqr_coef: float = attr.ib(default=1.5)
+    random_state: int = attr.ib(default=32)
+
+    @property
+    def keyname(self) -> str:
+        """
+        Compose a config's key name based on the iqr_coef and random_state parameters.
+
+        Returns
+        -------
+        str
+            String representing the key name in the format `"{iqr_coef}_{random_state}"`.
+        """
+        return f"{self.iqr_coef}_{self.random_state}"
+
+
+def make_downsample(interactions: pd.DataFrame, debias_config: DebiasConfig) -> pd.DataFrame:
+    """
+    Downsample the size of interactions, excluding some interactions with popular items.
+
+    Algorithm:
+
+        1. Calculate item popularity distribution from interactions;
+        2. Find first (Q1) and third (Q3) quartiles in items popularity distribution;
+        3. Calculate IQR = Q3 - Q1;
+        4. Calculate maximum value inside by formula: Q3 + iqr_coef * IQR;
+        5. Down-sample for all exceeding items in interactions,
+        randomly keeping the maximum group of users to a size not exceeding
+        maximum value inside
+
+    Parameters
+    ----------
+    interactions : pd.DataFrame
+        Table with previous user-item interactions,
+        with columns `Columns.User`, `Columns.Item`.
+    iqr_coef : float, default 1.5
+        Coefficient for defining as the maximum value inside the border.
+    random_state: float, default 32
+        Pseudorandom number generator state to control the down-sampling.
+
+    Returns
+    -------
+    pd.DataFrame
+        Downsampling interactions.
+    """
+    if len(interactions) == 0:
+        return interactions
+
+    item_popularity = interactions[Columns.Item].value_counts()
```

Comment:
What if there are multiple rows with the same user+item pairs?  Don't we want to use the same approach of popularity calculation we used in popularity models?

Reply:
It's a point to discuss.  I think now it's move correct.  But probably it'd be nice to have different options, and it also makes sense to keep the things in one style.  
@blondered wdyt?

Reply:
My main point: in all applicable for debias models we guarantee that multiple rows for the same user-item pair do not have effect on metric value. They are actually dropped (since we need them only to assign TP to reco pairs).
Because of this we can actually deduplicate before downsampling. And then take simple value_counts. It kinda guarantees correct downsampling.

Reply:
We've discussed that we need to skip 'popularity' word in all docstrings and code. It is kinda confusing. Instead let's call it "number of users that interacted with item". Let's please do it everywhere and just not mention "popularity" at all.

Reply:
In this PR. And you need to do this only for your new module. In other places we still have "popularity" and it is correct. We are changing the naming in docstrings for de-biasing in order to not mix train dataset "popularity" and test interactions "number of users that interacted with item" which directly affects all metrics computation.
We're also very open to discuss other formulations of this concept :)

Reply:
Fixed. Is it correct now?

Reply:
Yes, it is correct in the code, thank you! I proposed some changes to the docstrings below.
@feldlime will resolve this when he is happy :)